client/op.go

```diff
@@ -0,0 +1,490 @@
+package client
+
+import xlineapi "github.com/xline-kv/go-xline/api/xline"
+
+// OpOption configures Operations like Get, Put, Delete.
+type OpOption func(*Op)
+
+// Op represents an Operation that kv can execute.
+type Op struct {
+	t   opType
+	key []byte
+	end []byte
+
+	// for range
+	limit             int64
+	sort              *SortOption
+	serializable      bool
+	keysOnly          bool
+	cntOnly           bool
+	minModRev         int64
+	maxModRev         int64
+	minCreateRev      int64
+	maxCreateRev      int64
+	isOptsWithFromKey bool
+	isOptsWithPrefix  bool
+
+	// for put
+	val      []byte
+	ignVal   bool
+	ignLease bool
+
+	// txn
+	cmps    []Cmp
+	thenOps []Op
+	elseOps []Op
+
+	// for compact
+	physical bool
+
+	// for grant
+	ttl int64
+
+	// for TimeToLive
+	attachedKeys bool
+
+	// for range, compact
+	rev int64
+
+	// for range, put, delete
+	prevKV bool
+
+	// for put, grant, ttl
+	leaseID int64
+}
```

Comment:
This struct contains too many fields, consider the following refactoring.

```go
type Op struct {
	t   opType
	key []byte
	end []byte

	rangeOp *rangeOp

	putOp *putOp

	txnOp *txnOp
    
    ...
}

type struct rangeOp {
    ...
}

...
```

Reply:
Your idea are great. I will refactor as you suggested.